packages/fxa-settings/src/components/ButtonDownloadRecoveryKey/index.tsx

```diff
@@ -0,0 +1,126 @@
+  viewName: string;
+}
+
+export const ButtonDownloadRecoveryKey = ({
+  recoveryKeyValue,
+  viewName,
+}: ButtonDownloadRecoveryKeyProps) => {
+  // TODO: format by locale
+  const { primaryEmail } = useAccount();
+  const currentDate = new Date();
+  const downloadDateForFilename = currentDate.toISOString().split('T')[0];
+  const downloadDateInLocale = currentDate.toLocaleDateString(
+    navigator.language
+  );
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const fileHeading = ftlMsgResolver.getMsg(
+    'recovery-key-file-header',
+    'SAVE YOUR ACCOUNT RECOVERY KEY'
+  );
+
+  const fileInstructions = ftlMsgResolver.getMsg(
+    'recovery-key-file-instructions',
+    'Store this file containing your account recovery key in a place you’ll remember. Or print it and keep a physical copy. Your account recovery key can help you recover Firefox data if you forget your password.'
+  );
+
+  const fileKey = ftlMsgResolver.getMsg(
+    'recovery-key-file-key-value',
+    `🔑 Key:  ${recoveryKeyValue}`,
+    { recoveryKeyValue }
+  );
+
+  const fileUserEmail = ftlMsgResolver.getMsg(
+    'recovery-key-file-user-email',
+    `➤ Firefox account: ${primaryEmail.email}`,
+    { email: primaryEmail.email }
+  );
+
+  const fileDate = ftlMsgResolver.getMsg(
+    'recovery-key-file-download-date',
+    `➤ Key generated: ${downloadDateInLocale}`,
+    { downloadDate: downloadDateInLocale }
+  );
+
+  const supportURL = 'https://mzl.la/3bNrM1I';
+
+  const fileSupport = ftlMsgResolver.getMsg(
+    'recovery-key-file-support',
+    `➤ Learn more about your account recovery key: ${supportURL}`,
+    { supportURL }
+  );
+
+  const fileContent = new Blob(
+    [
+      fileHeading,
+      '\r\n\r\n',
+      fileInstructions,
+      '\r\n\r\n',
+      fileKey,
+      '\r\n\r\n',
+      fileUserEmail,
+      '\r\n',
+      fileDate,
+      '\r\n',
+      fileSupport,
+    ],
+    {
+      type: 'text/plain',
+    }
+  );
+
+  const truncateStringToSingleChar = (string: string, char: string) => {
+    const stringParts = string.split(char);
+    let truncatedStringParts: string[] = [];
+    stringParts.forEach((part) => truncatedStringParts.push(part[0]));
+    return truncatedStringParts.join(char);
+  };
+
+  const getFormattedFilename = () => {
+    const fullFilename = `Firefox_Account_Recovery_Key_${primaryEmail.email}_${downloadDateForFilename}.txt`;
+
+    // if the fullFilename is close to filesystem max, we truncate the email address
+    if (fullFilename.length > 200) {
+      const [username, domain] = primaryEmail.email.split('@');
+      const truncatedUsername = truncateStringToSingleChar(username, '.');
+      const truncatedDomain = truncateStringToSingleChar(domain, '.');
+      const truncatedEmail = `${truncatedUsername}@${truncatedDomain}`;
+      const truncatedFilename = `Firefox_Account_Recovery_Key_${truncatedEmail}_${downloadDateForFilename}.txt`;
+      return truncatedFilename;
+    } else {
+      return fullFilename;
+    }
```

Comment:
I'll happily defer to the eng team, but&hellip; it feels like truncating `"peter.dehaan+firefoxaccounts@gmail.co.uk"` to `"p.d@g.c.u"` might be a _bit_ excessive (so if the filename is over 200 characters long, we'll probably end up cutting out 90% of the string during truncation since we only take the first character after a "." delimiter).

What if we did something like truncate the email address in 5 character increments if the file is over a max length?
https://runkit.com/pdehaan/6441c29977019f0008cf4cec (and technically we wouldn't even need a loop if we know the suffix+prefix lengths; assuming we're using non-translated values, we would know the exact maximum length for the email portion and could do a simple slice).

So if the max length of the string is 100 characters, it'd shrink some absurdly long email address like `"peter.dehaan.is.a.fool.and.unlikeable.character@gmail.co.uk.biz"` to `"peter.dehaan.is.a.fool.and.unlikeable.character@gmail"` (and then with the prefix+suffix, the final string length is only 97 characters:

```js
{
  filename: "Firefox_Account_Recovery_Key_peter.dehaan.is.a.fool.and.unlikeable.character@gmail_2023-04-20.txt",
  maxLen: 100,
  len: 97
}
```

Although I'm really guessing we're dealing w/ edge cases here since if the max filename length is 200 characters and the prefix+suffix is 44 characters, that still leaves a maximum of 156 characters for the email address portion, and I'm doubting many people would have that in the prod database for a primary email address.

Reply:
I may have been a bit truncate-happy and... ahem... aggressively shortened the email. 🔪 I very much like your suggestion, and will swap my chopping with some more selective slicing.
PS - You should consider changing your address to `"peter.dehaan.is.a.wizard.and.always.has.really.good.suggestions@emails.for.wise.wizards.ca"`